data/volunteer.schema.json

```diff
@@ -0,0 +1,112 @@
+{
+	"$schema": "https://json-schema.org/draft-07/schema",
+	"$id": "https://vms.torontojs.com/volunteer.schema.json",
+	"title": "VMS Data Schema",
+	"description": "Description for data and fields saved by the VMS.",
+	"definitions": {
+		"agreement": {
+			"type": "object",
+			"description": "An agreement signed by the volunteer, if contains information about the date it was signed, as well as the document template version.",
+			"required": ["signedAt", "document", "version"],
+			"properties": {
+				"signedAt": {
+					"type": "string",
+					"format": "date-time",
+					"description": "A timestamp for when the agreement was signed."
+				},
+				"document": {
+					"type": "string",
+					"description": "The specific document template that the volunteer signed.",
+					"enum": ["VOLUNTEER_AGREEMENT", "IMAGE_RELEASE"]
+				},
+				"version": {
+					"type": "string",
+					"description": "The version of the document used when signing. This is used so when the documents update, we keep track of each document signed."
+				}
+			}
+		},
+		"team": {
+			"type": "object",
+			"description": "A team that the volunteer may be part of.",
+			"required": ["id", "joinedAt", "role"],
+			"properties": {
+				"id": {
+					"type": "string",
+					"format": "uuid",
+					"description": "The id for the team the volunteer joined."
+				},
+				"joinedAt": {
+					"type": "string",
+					"format": "date-time",
+					"description": "A timestamp representing when the volunteer joined the team."
+				},
+				"leftAt": {
+					"type": "string",
+					"format": "date-time",
+					"description": "A timestamp representing when the volunteer left the team."
+				},
+				"role": {
+					"type": "string",
+					"description": "The role the volunteer have for the team in question."
+				}
+			}
+		}
+	},
+	"type": "object",
+	"required": ["id", "email", "socialMedia", "joinedAt", "agreements"],
+	"properties": {
+		"id": {
+			"type": "string",
+			"format": "uuid",
+			"description": "A UUID identifying the volunteer internally. this field is generated when the data is saved."
+		},
+		"email": {
+			"type": "string",
+			"format": "email",
+			"description": "The volunteer's email address. This is the external identifier used for adding a volunteer to the system and to check for duplicates."
+		},
+		"name": {
+			"type": "string",
+			"description": "A name the volunteers would like to be identified by."
+		},
+		"pronouns": {
+			"type": "string",
+			"description": "The volunteer preffered pronouns, if they wish to provide it."
+		},
+		"socialMedia": {
+			"type": "object",
+			"description": "A list of social media links/handles for reaching out to the volunteer.",
+			"required": ["slack"],
+			"properties": {
+				"slack": { "type": "string", "format": "uri" },
```

Comment:
One idea is to call this 'links' instead of 'socialMedia', make it an array and give each item in the array the following properties:

* href
* title
* rel

Now it's a general links object, but you can still find out what 'social network' it was by looking at the domain for example. But it's completely unrestricted. `rel` should be `me` for most of those.

Reply:
That is a good structure! What about, if we only allow for `rel=me` that we have a flag `isRelMe` instead of an open attribute?

Are there other values for the `rel` attribute that would be interesting to have?

Reply:
I'm trying to sneak a hypermedia format in here :P But there's lots of others:

https://www.iana.org/assignments/link-relations/link-relations.xhtml

If you only want to support `me` I would probably recommend `"const": "me"`. If all of these are technically links to external profiles they are probably all `"me"`.

Reply:
But one could be "friend" from this list: https://microformats.org/wiki/existing-rel-values

Reply:
Technically, all would be links to that person...

Yeah, we can definitely make it more generic. I just think would be nice to have a strong default and offer extensions if needed.

Reply:
I like this idea 👍